packages/transactions/src/serializers/message.ts

```diff
@@ -9,34 +9,27 @@ import {
     VariableSizeDecoder,
     VariableSizeEncoder,
 } from '@solana/codecs-core';
-import {
-    getArrayDecoder,
-    getArrayEncoder,
-    getStructDecoder,
-    getStructEncoder,
-    StructToDecoderTuple,
-    StructToEncoderTuple,
-} from '@solana/codecs-data-structures';
+import { getArrayDecoder, getArrayEncoder, getStructDecoder, getStructEncoder } from '@solana/codecs-data-structures';
 import { getShortU16Decoder, getShortU16Encoder } from '@solana/codecs-numbers';
 import { getBase58Decoder, getBase58Encoder, getStringDecoder, getStringEncoder } from '@solana/codecs-strings';
 
-import { getCompiledAddressTableLookups } from '../compile-address-table-lookups';
+import type { getCompiledAddressTableLookups } from '../compile-address-table-lookups';
 import { CompiledMessage } from '../message';
 import { getAddressTableLookupDecoder, getAddressTableLookupEncoder } from './address-table-lookup';
 import { getMessageHeaderDecoder, getMessageHeaderEncoder } from './header';
 import { getInstructionDecoder, getInstructionEncoder } from './instruction';
 import { getTransactionVersionDecoder, getTransactionVersionEncoder } from './transaction-version';
 
 function getCompiledMessageLegacyEncoder(): VariableSizeEncoder<CompiledMessage> {
-    return getStructEncoder(getPreludeStructEncoderTuple());
+    return getStructEncoder(getPreludeStructEncoderTuple()) as VariableSizeEncoder<CompiledMessage>;
```

Comment:
Is the cast needed despite the return type already being specified?

Reply:
Yeah that's specific to the `CompiledMessage` type though. It's because `TransactionVersion` is of type `"legacy" | 0` whereas some nested type within the codec definition defines them as `number` so the cast is just for that.

We could probably improve some of the codec internals inside the transactions package but I wanted to make sure this PR didn't diverge into a big multi-package refactoring.